pylink/jlink.py

```diff
@@ -5191,3 +5191,39 @@ def cp15_register_write(self, cr_n, op_1, cr_m, op_2, value):
         if res != 0:
             raise errors.JLinkException(res)
         return res
+
+###############################################################################
+#
+# JTAG API (Raw JTAG access for boundary scan or other purposes)
+#
+###############################################################################
+
+    @interface_required(enums.JLinkInterfaces.JTAG)
+    def jtag_rawrw(self, tdo, tms, numbits=None):
+        """Writes a list of raw bits to TDO+TMS, returns TDI input.
+
+        Args:
+          self (JLink): the ``JLink`` instance
+          tdo (List): List of bytes to be shifted out tdo.
+          tms (List): list of bytes to be shifted out tms.
+          numbits (Int): Number of bits to shift out in total.
+
+        Returns:
+          tdi (List): Output from device shifted into tdi.
+        """
+
+        if len(tdo) != len(tms):
+            raise ValueError("TMS & TDO arrays must be same length")
+
+        buf_size = len(tdo)
+
+        if numbits is None:
+            numbits = len(tdo)*8
```

Comment:
Minor nit, but `len(tdo) * 8`; spaces around the operator.

Reply:
Oops - lazy Python here, fixed in updated PR.